apps/prairielearn/src/lib/workspace.js

```diff
@@ -397,14 +425,25 @@ module.exports = {
 
           // Discard encodings outside of explicit list of allowed encodings
           if (file.encoding && !['utf-8', 'base64', 'hex'].includes(file.encoding)) {
+            fileGenerationErrors.push({
+              file: file.name,
+              msg: `Dynamic workspace file has unsupported file encoding (${file.encoding}). File ignored.`,
+              data: file,
+            });
             return null;
           }
           return {
             name: file.name,
             buffer: Buffer.from(file.contents ?? '', file.encoding || 'utf-8'),
           };
-        } catch (_err) {
+        } catch (err) {
           // Error retrieving contents of dynamic file. Ignoring file.
+          fileGenerationErrors.push({
+            file: file.name,
+            msg: `Error decoding dynamic workspace file: ${err.message}`,
+            err,
+            data: { file },
```

Comment:
In pretty much all other lines, we do this:

```suggestion
            data: file,
```

I don't particularly care which we do, but I'd prefer to remain consistent, unless there's a good reason not to.

Reply:
Done 441eb4e.